java-checks/src/main/java/org/sonar/java/checks/ToStringUsingBoxingCheck.java

```diff
@@ -86,11 +89,15 @@ private static String getNewlyCreatedClassName(MethodInvocationTree mit) {
     return "";
   }
 
-  private static boolean isCallingToString(MethodInvocationTree mit) {
-    if (mit.methodSelect().is(Tree.Kind.MEMBER_SELECT)) {
-      MemberSelectExpressionTree mset = (MemberSelectExpressionTree) mit.methodSelect();
-      return "toString".equals(mset.identifier().name());
+  private static String isCallingToStringOrCompareTo(ExpressionTree methodSelect) {
```

Comment:
Even if it's a private method, I really think that you should add the `@CheckForNull` annotation.